test/puppetlabs/dujour/version_check_test.clj

```diff
@@ -0,0 +1,61 @@
+(ns puppetlabs.dujour.version-check-test
+  (:require [clojure.test :refer :all]
+            [puppetlabs.trapperkeeper.testutils.webserver :as jetty9]
+            [puppetlabs.trapperkeeper.testutils.logging :refer [with-test-logging]]
+            [puppetlabs.dujour.version-check :refer :all]
+            [schema.test :as schema-test]
+            [cheshire.core :as json]))
+
+(use-fixtures :once schema-test/validate-schemas)
+
+(deftest test-get-coords
+  (testing "group-id should use the default if not specified"
+    (is (= {:group-id    "puppetlabs.packages"
+            :artifact-id "foo"}
+           (get-coords "foo"))))
+  (testing "should use group-id if specified"
+    (is (= {:group-id    "foo.foo"
+            :artifact-id "foo"}
+           (get-coords {:group-id "foo.foo"
+                        :artifact-id "foo"})))))
+
+(defn update-available-app
+  [req]
+  {:status 200
+   :body (json/generate-string {:newer true
+                                :link "http://foo.com"
+                                :message "Howdy!"
+                                :product "foo"
+                                :version "9000.0.0"})})
+
+(defn server-error-app
+  [req]
+  {:status 500
+   :body "aaaaaaaaaaaaaaaaaaaaaaaaaa"})
+
+(deftest test-check-for-updates!
+  (with-test-logging
+    (jetty9/with-test-webserver update-available-app port
+                                (check-for-updates! "foo" (format "http://localhost:%s" port))
+                                (is (logged? #"Newer version 9000.0.0 is available!" :info))))
+  (with-test-logging
+    (jetty9/with-test-webserver server-error-app port
+                                (check-for-updates! "foo" (format "http://localhost:%s" port))
+                                (is (logged? #"Could not retrieve update information" :debug)))))
+
+(deftest test-version-check
+  (with-test-logging
+    (jetty9/with-test-webserver update-available-app port
+                                (version-check "foo" (format "http://localhost:%s" port))
+                                (Thread/sleep 1000)
```

Comment:
OK, so, here's my follow-up on the whole future-vs-agent question.

In the current state of this PR, I agree with you, @jeffmccune .  There's not a real advantage.

What I was envisioning, though, is something more similar to how we're using agents in the Puppet Server jruby stuff, to handle flush requests in the background.  In that scenario, what you'd do is create an agent during the initialization of the TK service, and then stick it into the service's context map so that it could be referenced later.  Then, tests could pluck the agent out of the context map, add a watch to it for test purposes, and then fire off the function call that would trigger the actual behavior that we want to test.  With a future you couldn't do this because you can't separate the creation of the future from the call that causes it to execute the code asynchronously.

What I failed to realize (because I'm dumb, and because I wasn't paying close enough attention when I added my drive-by comments to this PR), is that since this is no longer a trapperkeeper service, it doesn't have a lifecycle, and thus there is no "init" phase to use to construct the agent, and no way to reference it later.

This leaves us with a few options that I don't love:

1. Return the agent from the public function so that the caller can interact with it afterwards.  That seems like a weird API choice unless it was motivated by a use case other than testing.
2. Use a future instead of an agent, and return that from the public function.  This has the same semi-weird implications for the API.
3. Change the API to accept a callback function (as @fpringvaldsen has done).  This seems less weird as far as API goes, but still a little weird unless we can think of meaningful use cases beyond testing.
4. Revert those last few commits and just deal with the fact that we're going to have to sleep in the tests for now.

I'd probably vote for option 3 or option 4.

@fpringvaldsen this is pretty low priority, so don't worry about spending any more time on the code until we've come around to what we think the final implementation details should be... and @jeffmccune no rush on responding to any of this :)